src/test/ui/lint/reasons-erroneous.rs

```diff
@@ -0,0 +1,16 @@
+#![warn(absolute_paths_not_starting_with_crate, reason = 0)]
+//~^ ERROR malformed lint attribute
+//~| HELP reason must be a string literal
+#![warn(anonymous_parameters, reason = b"consider these, for we have condemned them")]
+//~^ ERROR malformed lint attribute
+//~| HELP reason must be a string literal
+#![warn(bare_trait_objects, reasons = "leaders to no sure land, guides their bearings lost")]
+//~^ ERROR malformed lint attribute
+#![warn(box_pointers, blerp = "or in league with robbers have reversed the signposts")]
+//~^ ERROR malformed lint attribute
+#![warn(elided_lifetimes_in_paths, reason("disrespectful to ancestors", "irresponsible to heirs"))]
+//~^ ERROR malformed lint attribute
+#![warn(ellipsis_inclusive_range_patterns, reason)]
```

Comment:
Could you add a couple more cases:
`#[warn(lint1, reason = "zzz", reason = "yyy")]` - duplicated reason (probably an error).
`#[warn(lint1, reason = "zzz", lint2)]` - reason is not the last (an error? just to be conservative)

Reply:
> reason is not the last (an error? just to be conservative)

I think this is compliant with the spirit of the RFC? @Centril @myrrlyn

This is also nice from an implementer's perspective—in the first revision of this PR, I felt bad about doing a entire extra iteration over the meta items just to pick up the reason before entering the loop that processes the lint names themselves, but with this restriction, we can just peek at the end.